plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/mutate/MutateProcessor.java

```diff
@@ -0,0 +1,312 @@
+                oldVal = false;
+            } else {
+                throw new IllegalArgumentException("Cannot convert value: " + oldVal.toString() + "into a boolean");
+            }
+        }
+        return oldVal;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void doConvert(Data data) {
+        if (convert == null) { return; }
+
+        for(Map.Entry<String, String> entry : convert.entrySet()) {
+            String toType = entry.getValue();
+
+            Object oldVal = data.getProperty(entry.getKey());
+            Object newVal;
+
+            if (oldVal instanceof List) {
+                newVal = new ArrayList<>();
+                for (Object e : ((List<Object>) oldVal)) {
+                    ((List<Object>) newVal).add(parseValueAsType(e, toType));
+                }
+            } else {
+                newVal = parseValueAsType(oldVal, toType);
+            }
+
+            data.addField(entry.getKey(), newVal);
+        }
+    }
+
+    private void doSplit(Data data) {
+        if (split == null) { return; }
+
+        for(Map.Entry<String, String> entry : split.entrySet()) {
+            Object oldVal = data.getProperty(entry.getKey());
+            if (oldVal instanceof String) {
+                data.addField(entry.getKey(), Arrays.asList(((String) oldVal).split(entry.getValue())));
+            } else {
+                throw new IllegalArgumentException("Cannot split a field that is not a String type");
+            }
+        }
+    }
+
+    private void doGsub(Data data) {
+        if (gsub == null) { return; }
+
+        for (Map.Entry<String, List<String>> entry : gsub.entrySet()) {
+            String fieldName = entry.getKey();
+            List<String> matchAndReplaceList = entry.getValue();
+            String oldVal = data.getProperty(fieldName);
+            String newVal = oldVal.replaceAll(matchAndReplaceList.get(0), matchAndReplaceList.get(1));
+            data.addField(entry.getKey(), newVal);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void doJoin(Data data) {
+        if (join == null) { return; }
+
+        for(Map.Entry<String, String> entry : join.entrySet()) {
+            Object oldVal = data.getProperty(entry.getKey());
+            if (oldVal instanceof List) {
+                String joined = (String) ((List) oldVal)
+                        .stream()
+                        .map(Object::toString)
+                        .collect(Collectors.joining(entry.getValue()));
+
+                data.addField(entry.getKey(), joined);
+            } else {
+                throw new IllegalArgumentException("Cannot join field:" + entry.getKey() + " with type: " + oldVal.getClass());
+            }
+        }
+    }
+
+    private void doRemove(Data data) {
+        if (remove == null) { return; }
+        for(String field : remove) {
+            data.getDocument().remove(field);
+        }
+    }
+
+    private void doStrip(Data data) {
```

Comment:
I was also annoyed by the same thing. I left it as `strip` to keep it in line with `logstash-filter-mutate`, but I am good with moving it to trim since there is no tangible dependency on logstash semantics.